server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -609,16 +609,22 @@ public String toString() {
 
         private abstract class TimeUnitPreparedRounding extends PreparedRounding {
             @Override
-            public double roundingSize(long utcMillis, DateTimeUnit timeUnit) {
+            public double roundingSize(Long utcMillis, DateTimeUnit timeUnit) {
```

Comment:
I wonder if it'd be easier to read if there were two methods? I'd kind of feel more comfortable having two just so you don't need auto-boxing. It might not make a huge difference, but in general `Rounding` doesn't its best not to allocate stuff. I think this method doesn't have to be so efficient, but still.